src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -38,6 +43,8 @@ public abstract class AbstractLanguageServer<TRequestContext> : ILifeCycleManage
         _jsonRpc = jsonRpc;
         _jsonRpc.AddLocalRpcTarget(this);
         _jsonRpc.Disconnected += JsonRpc_Disconnected;
+        _lspServices = new Lazy<ILspServices>(() => ConstructLspServices());
+        _queue = new Lazy<IRequestExecutionQueue<TRequestContext>>(() => ConstructRequestExecutionQueue());
```

Comment:
@dibarbet (cc @mgoertz-msft ) this change broke XAML Language service. Virtual method ConstructRequestExecutionQueue is now called from within [RoslynLanguageServer](https://github.com/dotnet/roslyn/blob/66fc6275b54428f2b5536b867392097c9426038b/src/Features/LanguageServer/Protocol/RoslynLanguageServer.cs#L42) constructor, which calls Initialize > GetRequestExecutionQueue > _queue.Value. Net result is that [XamlLanguageServer.ConstructRequestExecutionQueue](https://github.com/dotnet/roslyn/blob/66fc6275b54428f2b5536b867392097c9426038b/src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/XamlLanguageServer.cs#L42) is called too early, when fields are not initialized. We end up with bad XamlRequestExecutionQueue.

Reply:
ick.  we really need to seal our types.  subclassing concrete types is not a way that roslyn considers that extensibility should be allowed.